File: Frontend/src/app/shared/models/form-request.ts
export interface FormRequest {
  id: string;
  name: string;
  owner: string;
  icon: string;
  created: string;
  description: string;
  color: string;
  hideSections: boolean;
  sections: FormSection[]
}

export interface FormSection {
  id: string;
  name: string;
  questions: FormQuestion[];
}
  
export interface FormQuestion {
  id: string;
  text: string;
  type: 'SingleChoice' | 'MultipleChoice' | 'Text' | 'YesNo' | 'Date' | 'Dropdown' | 'AD' | 'Number';
  options: string[];
  required: boolean;
  minAnswer: number;
  maxAnswer: number;
  conditionalVisibilities: ConditionalVisibility[];
}

export interface FormQuestionNoConditionals { //Strictly for COnditional Visibiltiy Selection
  id: string;
  text: string;
  type: 'SingleChoice' | 'MultipleChoice' | 'Text' | 'YesNo' | 'Date' | 'Dropdown' | 'AD' | 'Number';
  options: string[];
}


export interface ConditionalVisibility {
  sectionId: string;
  sectionName: string;
  questionId: string;
  questionText: string;
  type: 'Equals' | 'NotEquals' | 'Contains' | 'NotContains';
  option: string;
  numberOption: number;
  options: string[];
  numberOptions: number[];
  possibleQuestions?: FormQuestionNoConditionals[]; //Strictly for COnditional Visibiltiy Selection
  isNumberCond?: boolean;
}

File: Service/FormAdvanced.Domain/Entities/FormRequest.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FormAdvanced.Domain.Entities
{

    public record SubmittedRequest(
        string id,
        string FormName,
        string FormId,
        string ApproverUID,
        string ApproverName,
        string Owner,
        string OwnerName,
        string State,
        string Icon,
        DateTime Created,
        string Description,
        string Color,
        bool HideSections,
        SubmittedSection[] Sections
    );

    public record SubmittedSection(
        string SectionId,
        string Name,
        SubmittedRequestQuestion[] Questions
    );

    public record SubmittedRequestQuestion
    (
        string QuestionId,
        string Text,
        string Type,
        bool Required,
        string[] Options,
        string Answer,
        string[] Answers,
        int? MinAnswer,
        int? MaxAnswer,
        SubmittedRequestConditionalVisibility[] ConditionalVisibilityTriggerForOtherQuestion
    );

    public record SubmittedRequestConditionalVisibility
    (
        string SectionId,
        string SectionName,
        string QuestionId,
        string QuestionText,
        string Type,
        string Option,
        int NumberOption,
        string[] Options,
        int[] NumberOptions
    );
}